axelrod/strategies/lookerup.py

```diff
@@ -165,60 +158,62 @@ def strategy(self, opponent):
         return action
 
 
-# Create several classes at runtime based on the data in the look up tables
-# loaded above, one for each key.
+class EvolvedLookerUp1_1_1(LookerUp):
+    """
+    A 1_1_1 Lookerup trained with an evolutionary algorithm.
+
+    Names:
+        - Evolved Lookerup 1 1 1: Original name by Marc Harper
+    """
+    name = "EvolvedLookerUp1_1_1"
+    pattern = 'CDDDDDCD'
+    parameters = (1, 1, 1)
+    initial_actions = (C,)
+
+    def __init__(self):
+        plays, op_plays, op_start_plays = self.parameters
+        lookup_table = create_lookup_table_from_pattern(plays,
+                                                        op_plays,
+                                                        op_start_plays,
+                                                        pattern=self.pattern)
+        super().__init__(lookup_table=lookup_table,
+                          initial_actions=self.initial_actions)
+
+
+class EvolvedLookerUp2_2_2(EvolvedLookerUp1_1_1):
+    """
+    A 2_2_2 Lookerup trained with an evolutionary algorithm.
 
-for k, (initial, pattern) in patterns.items():
-    name, plays, op_plays, op_start_plays = k
-    table = create_lookup_table_from_pattern(
-        plays, op_plays, op_start_plays, pattern)
-    class_name = "EvolvedLookerUp{}{}_{}_{}".format(
-        name, plays, op_plays, op_start_plays)
-    # Dynamically create the class
-    new_class = type(class_name, (LookerUp,), {})
-    new_class.__init__ = init_args(partial(
-        new_class.__init__, lookup_table=table, initial_actions=initial))
-    new_class.name = class_name
-    # Add the generated class to the module dictionary so it can be
-    # imported into other modules
-    setattr(module, class_name, new_class)
+    Names:
+        - Evolved Lookerup 2 2 2: Original name by Marc Harper
+    """
+    name = "EvolvedLookerUp2_2_2"
+    pattern = 'CDCCDCCCDCDDDCCCDCDDDDDDDCDDDCDCDDDDCCDCCCCDDDDCCDDDDCCDCDDDDDDD'
+    parameters = (2, 2, 2)
+    initial_actions = (C, C)
 
 
-class Winner12(LookerUp):
```

Comment:
I prefer inheriting from LookerUp to reduce the number of inherited classes. It seems odd to me that Winner12 would derive from EvolvedLookerUp1_1_1.